## Local regression tests failed

The implement subagent opened this PR even though the local `python -m unittest` run failed after commit. The failures below are posted here as a reviewer finding so `cai-revise` picks them up and addresses them on the PR side rather than rolling the issue back through another full implement cycle.

### Failing tests

```
FAIL: test_planned_can_fall_back_to_human (test_fsm.TestTransientStatesShape.test_planned_can_fall_back_to_human)
PLANNED → PLAN_APPROVED is confidence-gated; explicit human path too.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/cai-implement-1167-5592bbb3/tests/test_fsm.py", line 724, in test_planned_can_fall_back_to_human
    self.assertEqual(dests, {IssueState.PLAN_APPROVED, IssueState.HUMAN_NEEDED})
AssertionError: Items in the first set but not the second:
<IssueState.SPLITTING: 'auto-improve:splitting'>

FAILED (failures=1, skipped=1)
```

---
_Posted by `cai implement`. Addressing this comment clears the finding; the PR continues through review → merge as usual._

--- a/docs/fsm.md
+++ b/docs/fsm.md
@@ -76,6 +76,7 @@ stateDiagram-v2
   PLANNING --> HUMAN_NEEDED: planning_to_human [≥HIGH]
   PLANNED --> PLAN_APPROVED: planned_to_plan_approved [≥HIGH] | planned_to_plan_approved_mitigated [≥MEDIUM] | planned_to_plan_approved_docs_only [≥MEDIUM] | planned_to_plan_approved_approvable [≥MEDIUM]
   PLANNED --> HUMAN_NEEDED: planned_to_human [≥HIGH]
+  PLANNED --> SPLITTING: planned_to_splitting [caller-gated]
   PLAN_APPROVED --> IN_PROGRESS: approved_to_in_progress [≥HIGH]
   PLAN_APPROVED --> REFINED: plan_approved_to_refined [caller-gated]
   IN_PROGRESS --> PR: in_progress_to_pr [≥HIGH]

--- a/docs/modules/tests.md
+++ b/docs/modules/tests.md
@@ -1,6 +1,6 @@
 # tests
 
-Pytest suite covering the FSM, dispatcher, handlers, helpers,
+unittest suite covering the FSM, dispatcher, handlers, helpers,
 parse, publish, rescue, transcript sync, and lint hygiene. Every
 functional module has at least one dedicated test module here;
 the suite is the primary safety net for refactors.
@@ -11,6 +11,7 @@ the suite is the primary safety net for refactors.
 - [`tests/test_dispatcher.py`](../../tests/test_dispatcher.py) —
   state→handler registries and routing.
 - [`tests/test_plan.py`](../../tests/test_plan.py),
+  [`tests/test_split.py`](../../tests/test_split.py),
   [`tests/test_multistep.py`](../../tests/test_multistep.py),
   [`tests/test_implement_consecutive_failures.py`](../../tests/test_implement_consecutive_failures.py),
   [`tests/test_implement_helper_extract.py`](../../tests/test_implement_helper_extract.py),
@@ -47,7 +48,7 @@ the suite is the primary safety net for refactors.
   **audit**, **transcripts**, **github-glue** — the suite covers
   every functional module.
 - Run by **cli** — `cmd_test` (`cai test`) is the wrapper that
-  installs deps and runs pytest + ruff.
+  installs deps and runs unittest + ruff.
 - Run by **installer** — `entrypoint.sh` runs `cai test` on fresh
   container starts in some deployment modes.
 - No reverse imports — nothing in the pipeline imports `tests/`.

--- a/tests/test_fsm.py
+++ b/tests/test_fsm.py
@@ -715,13 +715,14 @@ class TestTransientStatesShape(unittest.TestCase):
         self.assertIn(IssueState.HUMAN_NEEDED, dests)
 
     def test_planned_can_fall_back_to_human(self):
-        """PLANNED → PLAN_APPROVED is confidence-gated; explicit human path too."""
+        """PLANNED → PLAN_APPROVED is confidence-gated; explicit human path too.
+        Post-plan re-split (#1167) adds PLANNED → SPLITTING via planned_to_splitting."""
         dests = {
             t.to_state
             for t in ISSUE_TRANSITIONS
             if t.from_state == IssueState.PLANNED
         }
-        self.assertEqual(dests, {IssueState.PLAN_APPROVED, IssueState.HUMAN_NEEDED})
+        self.assertEqual(dests, {IssueState.PLAN_APPROVED, IssueState.HUMAN_NEEDED, IssueState.SPLITTING})
 
     def test_refined_advances_to_splitting_or_planning(self):
         """REFINED is a waypoint — next stop is either SPLITTING

